docs/2.10.0/docs/canton/usermanual/contract_keys.rst

```diff
@@ -272,14 +272,14 @@ The main difference from the ``Orchestrator`` contract is that the ``Generate``
 Caveats to keep in mind are:
 
 - Your application must ensure that you only ever create one ``Generator`` contract (e.g., by creating one when initializing the application for the first time).
-- All commands that create the ``Keyed`` contract must be issued by the maintainer (in particular, do not delegate choices on the ``Generator`` contract to other parties).
+- All commands that create the ``Keyed`` contract must be issued by the maintainer. In particular, do not delegate choices on the ``Generator`` contract to other parties, because those parties may not have full visibility into all the ``Keyed`` contracts and ``lookupByKey`` may therefore wrongly return ``None``.
 - You must not create ``Keyed`` contracts by any other means other than exercising the ``Generate`` choice.
 - The ``Generate`` choice as shown above will not abort the command if the contract with the given key already exists, it will just return the existing contract.
   However, this is easy to change.
 - This approach relies on a particular internal behavior of Canton (as discussed below).
   While we don't expect the behavior to change, we do not currently make strong guarantees that it will not change.
-- If the participant is connected to multiple sync domains, the approach may fail in future versions of Canton.
-  To be future-proof, you should only use it in the settings when your participant is connected to a single sync domain.
+- If the participant is connected to multiple sync domains in future versions of Canton, the ``Generator`` contract will not guarantee uniqueness, as explained below.
```

Comment:
```suggestion
- If the participant is connected to multiple sync domains in future versions of Canton, the ``Generator`` contract cannot guarantee uniqueness, as explained below.
```
We don't use future tense versions of 'will' per the Google developer style guide.